salesagility/SuiteCRM
# Fixing tests

## Description




- Fixing/updating tests.
- Fixing PHP errors (notices, warnings etc.) log errors instead..
- State checking updated: Test check system state at before/after classes instead test-methods. (makes travis faster) In dev-mode tests run before/after test-methods

documentation added: salesagility/SuiteDocs#78

## Motivation and Context


Stateless and unreliable tests

## How To Test This


- Install SuiteCRM (do not login, just get a fresh instance):
- ensure your composer is updated 'composer update'
- run command: `vendor/bin/codecept run unit -f --steps -vvv` 
(if you get a failure, switch on `$sugar_config['developerMode'] = true;` in your `config_override.php` and test again to see more details)

**advanced testing:**

- Make same test and change the system state in it. (with fake errors for e.g: change database, superglobals, files etc...)
- Switch on developer mode
- Run tests >> test process should failing at wrong test method
- Switch off developer mode
- Run tests >> test process should failing after wrong test class
- Fix the wrong test
- Run tests >> test process should passing

## Types of changes


### Final checklist
